frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { Dropdown } from '@patternfly/react-core/dist/js/components/Dropdown/Dropdown';
+import { DropdownItem } from '@patternfly/react-core/dist/js/components/Dropdown/DropdownItem';
+import { DropdownToggle } from '@patternfly/react-core/dist/js/components/Dropdown';
+import { CaretDownIcon } from '@patternfly/react-icons';
+
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+
+import { cephBlockPoolResource } from '../constants/resources';
+
+export const PoolResourceComponent: React.FC<PoolParams> = ({ onParamChange }) => {
+  const [poolData, poolDataLoaded, poolDataLoadError] = useK8sWatchResource<K8sResourceKind[]>(
+    cephBlockPoolResource,
+  );
+  const [isOpen, setOpen] = React.useState<boolean>(false);
+  const [poolName, setPoolName] = React.useState<string>(null);
+
+  const handleDropdownChange = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    const name = e.currentTarget.id;
+    setPoolName(name);
+    onParamChange(name);
+  };
+
+  const poolDropdownItems = poolData?.map((pool, key) => (
+    <DropdownItem
+      key={key}
+      component="button"
+      id={pool?.metadata?.name}
+      onClick={handleDropdownChange}
+    >
+      {pool?.metadata?.name}
+    </DropdownItem>
+  ));
+
+  return (
+    <>
+      {poolDataLoaded && !poolDataLoadError && (
```

Comment:
So if the fetch fails we show nothing ? Dont you want to show some kind of error message ?

Reply:
ack, add error message for this.